### PR TITLE
Added a strictArrayCreation formats flag to disable automatic creatio…

### PR DESCRIPTION
**Pull request by caturner81**
Creates a new formats flag that disables the automatic creation of empty arrays if a key is not provided.

case class Example(arr: Array[Int])

val ex = parse("{}").extract[Example]

Without the flag, ex will = Example(arr = Array())
With the flag, extract will fail

### CODE REVIEW

**Comment by fernandomora** [comment]
+1
I have previously had problems with this behavior.
I would expect to fail parsing if Jsons for the given example does not have at least an empty array:
{"arr":[]}

File: core/src/main/scala/org/json4s/Formats.scala [hunk 1]
```diff
@@ -49,6 +49,7 @@ trait Formats extends Serializable { self: Formats =>
   def companions: List[(Class[_], AnyRef)] = Nil
   def allowNull: Boolean = true
   def strictOptionParsing: Boolean = false
+  def strictArrayCreation: Boolean = false
```
Comment (seratch) [inline core/src/main/scala/org/json4s/Formats.scala]: A bit long name but how about `automaticEmptyArrayCreationWhenKeyAbsent` (default: true)?

File: tests/src/test/scala/org/json4s/StrictArrayCreationSpec.scala [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    }
+    "successfully parse a non empty array" in {
+      val model = parse(containsNonEmptyArray).extract[SingleArrayModel]
+      model.array must_== Array(1, 2)
+    }
+  }
+
+}
+
+case class SingleArrayModel(array: Array[Int])
```
Comment (seratch) [inline tests/src/test/scala/org/json4s/StrictArrayCreationSpec.scala]: I wonder if your implementation works fine when the `array` has default value. IMO, if the array has default value, MappingException should not be thrown even if the key is absent.

**Comment by jozic** [comment]
this is a possible fix for the long standing  #198, which is an issue many of us were hit by

@caturner81 do you plan on finishing it or should we start over?

**Comment by jozic** [comment]
should be closed now as #486 is merged

**Pull request closed without merging**